### ISSUE TITLE
[Bug Report] `ttnn.conv2d` failed with `buffer size must be divisible by new page size`

### ISSUE THREAD

**Issue by jerrysky3** [tenstorrent/tt-metal#15482]
**Describe the bug**
`ttnn.conv2d` failed with `buffer size must be divisible by new page size with:

- Input shape: (1, 38, 38, 728)
- Kernel shape: (728, 728, 1, 1)
- Stride: (2, 2)
- Padding: (0, 0)
- Dilation: (1, 1)
- Groups: 1

**To Reproduce**

Run the example below:

```py
import ttnn
import torch

def main(device):
    torch_input = torch.rand((1, 38, 38, 728), dtype=torch.bfloat16)
    torch_weight = torch.rand((728, 728, 1, 1), dtype=torch.bfloat16)

    stride = 2
    padding = 0
    dilation = 1
    groups = 1

    input_tensor = ttnn.from_torch(torch_input, layout=ttnn.TILE_LAYOUT, device=device)
    weight_tensor = ttnn.from_torch(torch_weight, layout=ttnn.ROW_MAJOR_LAYOUT)

    output_tensor, _, _, _, _ = ttnn.conv2d(
        input_tensor=input_tensor,
        weight_tensor=weight_tensor,
        batch_size=torch_input.shape[0],
        in_channels=torch_input.shape[3],
        out_channels=torch_weight.shape[0],
        input_height=torch_input.shape[1],
        input_width=torch_input.shape[3],
        kernel_size=(torch_weight.shape[2], torch_weight.shape[3]),
        stride=(stride, stride),
        padding=(padding, padding),
        dilation=(dilation, dilation),
        groups=groups,
        device=device,
    )

if __name__ == "__main__":
    try:
        device = ttnn.open_device(device_id=0, l1_small_size=16384)
        main(device)
    finally:
        ttnn.close_device(device)
```

**Please complete the following environment information:**
- OS: Ubuntu 20.04
- Version of software: 2aa18695fe4a275647edef372409732359f6ec49
- Card: N300

**Comment by mywoodstock** [comment]
@jerrysky3 could you try this again, it should be resolved.

**Comment by jerrysky3** [comment]
It passed now, thanks!
